src/gadgets/wnn.rs

```diff
@@ -141,37 +144,34 @@ impl<F: PrimeFieldBits> WnnInstructions<F> for WnnChip<F> {
         );
 
         // Assign the inputs to the first column of the hash chip
-        let mut permuted_inputs = vec![];
-        layouter.assign_region(|| "permuted inputs", |mut region| {
-            for (i, input) in inputs.iter().enumerate() {
-                let input_cell = region.assign_advice(
-                    || format!("input {}", i),
-                    self.config.bit2num_chip_config.input,
-                    i,
-                    || Value::known(F::from(*input as u64)),
-                )?;
-
-                permuted_inputs.push(input_cell);
-            }
-            Ok(())
-        })?;
+
+        let permuted_inputs = layouter.assign_region(
```

Comment:
This caused the failing test!

I'm not exactly sure how, but it has something to do with the callback of `assign_region` being called multiple times. In some passes, `region.assign_advice()` returns cells with a `None` value, and in others it's the actual value. Assigning `permuted_inputs` like this makes sure it actually contains the values when it needs to.

Reply:
That is a strange one, but good that it works now. Will keep in mind for the future